05_stringer/willshen8/stringer_test.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := "127.0.0.1\n"
+	actual := buf.String()
+
+	assert.Equal(t, expected, actual, "Test Failed in main! Actual is not equal to expected result")
+}
+
+func TestStringer(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    IPAddr
+		expected string
+	}{
+		{name: "Empty IP Address", input: IPAddr{0, 0, 0, 0}, expected: "0.0.0.0"},
+		{name: "Broadcast IP Address", input: IPAddr{255, 255, 255, 255}, expected: "255.255.255.255"},
+	}
+
+	for _, test := range tests {
+		expected := test.expected
```

Comment:
get rid of ```expected := test.expected```, use ```test.expected``` directly

Reply:
Moving tests as global var seems to resolve my problem here, I had a lint warning, hence I had expected := test.expected